src/lib/ppx_coda/versioned.ml

```diff
@@ -0,0 +1,114 @@
+(* versioned.ml -- static enforcement of versioned types via ppx
+
+   1) check that versioned type always in Stable.Vn.T module hierarchy
+   2) versioned types depend only on other versioned types (or OCaml built-in types)
+
+  to use, add coda_ppx to the dune pps list, and add "versioned" to deriving list for a type
+
+ *)
+
+open Core_kernel
+open Ppxlib
+
+let is_versioned_ident item =
+  match item with
+  | Pexp_ident id -> (
+    match id.txt with Lident s -> String.equal s "versioned" | _ -> false )
+  | _ -> false
```

Comment:
I think ppx_deriving also accepts a `Pexp_apply` with a single argument (see [here](https://github.com/ocaml-ppx/ppx_deriving/blob/master/src/api/ppx_deriving.cppo.ml#L500)), it would be good to handle that case here too.